web-ui/src/main/resources/catalog/components/admin/uiconfig/UiConfigDirective.js

```diff
@@ -109,13 +109,22 @@
       },
       link: function(scope, element, attrs) {
         element.val(JSON.stringify(scope.value));
-        element.on('change', function() {
-          var newValue = $(this).val();
-          try {
-            angular.merge(scope.value, JSON.parse(newValue));
-          } catch (e) {
-            console.warn('Error parsing JSON: ', newValue);
+
+        scope.$watch('value', function(newValue, oldValue, s) {
+          if (newValue !== oldValue) {
```

Comment:
If the content of the object has changed but the ref is the same this will evaluate to false, and as such will not trigger an update of the DOM input value. Or am I mistaken? (haven't tested)

Reply:
The actual code is 
```javascript
        scope.$watch('value', function(newValue, oldValue, s) {
          if (newValue !== oldValue) {
            element.val(JSON.stringify(newValue));
          }
        }, true);
```
The last `true` param makes it to do a deep compare, not only comparing the reference but the actual content.

Reply:
Yes, but isn't the deep compare pointless since `newValue !== oldValue` will return true only if the object ref has changed?

Reply:
Removed the check.